### DISCUSSION TITLE
Cannot specify list in FCST_PCP_COMBINE_INPUT_TEMPLATE

### DISCUSSION THREAD

**Discussion by gsketefian** [dtcenter/METplus#1696] [Configuration] [answered]
Hi,

I'm running `METplus-4.1.1` and `met 10.1.1` on NOAA's `Hera` HPC.

I am obtaining an error when specifying multiple comma-separated values for `FCST_PCP_COMBINE_INPUT_TEMPLATE` (i.e. when specifying this parameter to be an array).  I have the following line in my configuration file:
```
FCST_PCP_COMBINE_INPUT_TEMPLATE =
  {init?fmt=%Y%m%d%H}/mem1/postprd/{ENV[NET]}.t{init?fmt=%H}z.bgdawpf{lead?fmt=%HHH}.tm00.grib2,
  {init?fmt=%Y%m%d%H?shift=-43200}/mem2/postprd/{ENV[NET]}.t{init?fmt=%H?shift=-43200}z.bgdawpf{lead?fmt=%HHH?shift=43200}.tm00.grib2
```
This gives the following error in the log file:
```
07/08 16:32:13.769 metplus (command_builder.py:265) ERROR: (pcp_combine_wrapper.py:492) Could not find files to build accumulation in /scratch1/BMC/hmtb/beck/ens_design_RRFS/data using template
{init?fmt=%Y%m%d%H}/mem1/postprd/RRFSE_CONUS.t{init?fmt=%H}z.bgdawpf{lead?fmt=%HHH}.tm00.grib2,
{init?fmt=%Y%m%d%H?shift=-43200}/mem2/postprd/RRFSE_CONUS.t{init?fmt=%H?shift=-43200}z.bgdawpf{lead?fmt=%HHH?shift=43200}.tm00.grib2
```
I also tried setting `FCST_PCP_COMBINE_INPUT_TEMPLATE` explicitly, as follows:
```
FCST_PCP_COMBINE_INPUT_TEMPLATE =
  2021050800/mem1/postprd/RRFSE_CONUS.t00z.bgdawpf006.tm00.grib2,
  2021050712/mem2/postprd/RRFSE_CONUS.t12z.bgdawpf018.tm00.grib2
```
But this gave a similar error:
```
07/08 17:43:21.223 metplus (command_builder.py:265) ERROR: (pcp_combine_wrapper.py:492) Could not find files to build accumulation in /scratch1/BMC/hmtb/beck/ens_design_RRFS/data using template
2021050800/mem1/postprd/RRFSE_CONUS.t00z.bgdawpf006.tm00.grib2,
2021050712/mem2/postprd/RRFSE_CONUS.t12z.bgdawpf018.tm00.grib2
```
I am able to use the same template specification for the parameter `GEN_ENS_PROD_INPUT_TEMPLATE` in a different configuration file, i.e.
```
GEN_ENS_PROD_INPUT_TEMPLATE =
  {init?fmt=%Y%m%d%H}/mem1/postprd/{ENV[NET]}.t{init?fmt=%H}z.bgdawpf{lead?fmt=%HHH}.tm00.grib2,
  {init?fmt=%Y%m%d%H?shift=-43200}/mem2/postprd/{ENV[NET]}.t{init?fmt=%H?shift=-43200}z.bgdawpf{lead?fmt=%HHH?shift=43200}.tm00.grib2
```
and the `GenEnsProd` tool is able to parse it properly.  So I'm wondering why the `pcp-combine` case fails.  Maybe `FCST_PCP_COMBINE_INPUT_TEMPLATE` is not allowed to be an array?

Thanks for any help you can provide.

**Comment by georgemccabe** [comment, marked as answer]
Hi @gsketefian, You are correct that an array of templates isn't supported in FCST_PCP_COMBINE_INPUT_TEMPLATE, at least for configuring PcpCombine using the ADD method. The logic for the ADD method is somewhat specific to other existing use cases that assume the files gathered for the command will have valid times that shift backwards in time based on the accumulation interval contained in the files. The command you are attempting to build here appears to expect 2 files that are valid at the same time.

The PCPCombine tool can be configured in many different ways, so it is difficult to implement logic to properly handle all of these cases in a consistent way. Because of this, we have added an option to the PCPCombine wrapper to allow the user to define the command they want to build explicitly by setting `FCST_PCP_COMBINE_METHOD = USER_DEFINED`, which will allow you to explicitly define the command you would like to build by setting `FCST_PCP_COMBINE_COMMAND` to the full command to run (excluding the path to the pcp_combine executable). 

Here is an example use case that uses PCPCombine wrapper in this way:
https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/PCPCombine/PCPCombine_user_defined.html

If you are still having difficultly configuring METplus to run this use case, please provide us with the path to your configuration file on Hera along with a sample pcp_combine command that you are trying to build.

**Comment by gsketefian** [comment]
@georgemccabe Thanks for the explanation.  I will try out the `-derive` approach and get back to you if I have questions.
